pennylane/gradients/pulse_gradient.py

```diff
@@ -188,9 +224,14 @@ def _stoch_pulse_grad(tape, argnum=None, num_split_times=1, sampler_seed=None, s
             rule underlying the differentiation; also see details
         sample_seed (int): randomness seed to be used for the time samples in the stochastic
             parameter-shift rule
-        shots (None, int, list[int]): The device shots that will be used to execute the tapes outputted by this
-            transform. Note that this argument doesn't influence the shots used for tape execution, but provides
-            information about the shots.
+        shots (None, int, list[int]): The shots of the device used to execute the tapes which are
+            returned by this transform. Note that this argument does not *influence* the shots
+            used for execution, but informs the transform about the shots to ensure a compatible
+            return value structure.
```

Comment:
> informs the transform about the shots to ensure a compatible return value structure.

Still confused by this 😕 What is this parameter actually doing? Can you elaborate?

Reply:
It determines how the post-processing function should order the output axes: If a shot vector is used, the postprocessing function wants to know this, in order to put the shots axis at the right position. An exemplary function being used in `stoch_pulse_grad`, and other gradient transforms, is the helper function `_reorder_grad_axes_single_measure_shot_vector` [here](https://github.com/PennyLaneAI/pennylane/blob/fde1f24bd784d6ee2af5c980c2d5010b4c2bbe54/pennylane/gradients/parameter_shift.py#L337).

 I think the reason this is confusing to you is a very good reason: It is a dirty abstraction, currently, that the transform has to obtain this information explicitly like this.

Reply:
Oh wow 😄 I think this should have some sort of example in there then, something in the direction of "this is used to indicate when a shot vector is used. For example, if ... then shots needs to be .." ?

Reply:
Oh wow 😄 I think this should have some sort of example in there then, something in the direction of "this is used to indicate when a shot vector is used. For example, if ... then shots needs to be .." ?

Reply:
Yeah, I suppose that would be nice to have. Should we address this separately, though? It is in particular something that can be merges still next week, and it concerns other transforms :)

Reply:
sounds good 👍 please create a story / note to self so we dont forget about it 😅

Reply:
Oh, alright, that's great to hear :)